tests/virt_autotest/esxi_open_vm_tools.pm

```diff
@@ -0,0 +1,218 @@
+        while ($num && console('svirt')->run_cmd("! ping -c1 $vm_ip")) {
+            sleep(10);
+            $num--;
+        }
+    }
+    # Guest Power On
+    if (esxi_vm_power_getstate($vm_id) =~ m/Powered off/) {
+        record_info('Guest Power On');
+        esxi_vm_power_on($vm_id);
+        my $num = 6;
+        while ($num && console('svirt')->run_cmd("ping -c1 $vm_ip")) {
+            sleep(10);
+            $num--;
+        }
+    }
+    # Guest Reset
+    if (esxi_vm_power_getstate($vm_id) =~ m/Powered on/) {
+        record_info('Guest Reset');
+        esxi_vm_power_reset($vm_id);
+        my $num = 6;
+        while ($num && console('svirt')->run_cmd("ping -c1 $vm_ip")) {
+            sleep(10);
+            $num--;
+        }
+    }
+    # Guest Restart
+    if (esxi_vm_power_getstate($vm_id) =~ m/Powered on/) {
+        record_info('Guest Restart');
+        esxi_vm_power_reboot($vm_id);
+        my $num = 6;
+        while ($num && console('svirt')->run_cmd("ping -c1 $vm_ip")) {
+            sleep(10);
+            $num--;
+        }
+    }
+    # Guest Shutdown
+    if (esxi_vm_power_getstate($vm_id) =~ m/Powered on/) {
+        record_info('Guest Shutdown');
+        esxi_vm_power_shutdown($vm_id);
+        my $num = 6;
+        while ($num && console('svirt')->run_cmd("! ping -c1 $vm_ip")) {
+            sleep(10);
+            $num--;
+        }
+    }
+}
+
+sub networking_tests {
+    my ($vm_id, $vm_ip) = @_;
+
+    if (esxi_vm_power_getstate($vm_id) =~ m/Powered off/) {
+        esxi_vm_power_on($vm_id);
+        my $num = 6;
+        while ($num && console('svirt')->run_cmd("ping -c1 $vm_ip")) {
+            sleep(10);
+            $num--;
+        }
+    }
+
+    my $vswitch_name = esxi_vm_network_binding($vm_id);
+    record_info('VSwitch Name', $vswitch_name);
+
+    reset_consoles;
+    select_console("sut");
+    assert_screen("grub2", 90);
+    send_key 'ret';
+    select_console("root-console", timeout => 180);
+
+    assert_script_run("ping -c3 openqa.suse.de");
+    assert_script_run("ping -c3 www.suse.com");
+}
+
+sub clock_sync_tests {
+    my ($vm_name, $vm_id, $vm_ip) = @_;
+    my ($g_datetime, $h_datetime, $g_difftime);
+
+    # Enable timesync service in guest VM
+    if (script_output("vmware-toolbox-cmd timesync status", proceed_on_failure => 1, timeout => 90) == "Disabled") {
+        $g_datetime = script_output("date -u +'\%Y-\%m-\%d \%H:\%M:\%S'");
+        record_info('Guest current time before syncing up', $g_datetime);
+        assert_script_run("vmware-toolbox-cmd timesync enable", timeout => 90);
+        $h_datetime = get_host_timestamp();
+        record_info('Host time', $h_datetime);
+        $g_datetime = script_output("date -u +'\%Y-\%m-\%d \%H:\%M:\%S'");
+        record_info('Guest time after syncing up with host', $g_datetime);
+    }
+
+    my $h_timesec = get_timelc($h_datetime);
+    my $g_timesec = get_timelc($g_datetime);
+    my $diff_sec  = int($g_timesec - $h_timesec);
```

Comment:
How fast is the clock syncing? From code, after enabling clock sync, the host and guest time are retrieved one by one immediately, without dedicated time left for the syncing. Is this stable?

Reply:
As tested from my local openqa, it will sync immediately once enabling the timesync service, and it's stable enough. But anyway we could set a time on guest before retrieving which different from the host.